packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts

```diff
@@ -267,16 +282,20 @@ export class ZoweDatasetNode extends ZoweTreeNode implements IZoweDatasetTreeNod
             options.attributes = true;
             let label: string;
             if (contextually.isSessionNotFav(this)) {
-                this.pattern = this.pattern.toUpperCase();
-                // loop through each pattern for datasets
-                for (const pattern of this.pattern.split(",")) {
-                    responses.push(
-                        await ZoweExplorerApiRegister.getMvsApi(cachedProfile).dataSet(pattern.trim(), {
-                            attributes: true,
-                        })
-                    );
-                }
-            } else if (this.memberPattern !== undefined) {
+                responses.push(
+                    await ZoweExplorerApiRegister.getMvsApi(cachedProfile).dataSetsMatchingPattern(
```

Comment:
In the case that an Imperative error is encountered, I believe that `dataSetsMatchingPattern` will run much slower because it has to list data set attributes one-by-one instead of as a bulk operation. It would be nice if we can retrieve the list of data set names ASAP to populate the tree, and then fetch attributes as a background task and update the tree as needed.

Not suggesting a change in this PR since it would probably involve significant effort and maybe some updates to the Zowe SDK as well. Should we file an issue for a future enhancement? 🙂

Reply:
I believe we use the attributes to ultimately give the node a proper icon path.
I hope we don't have to refactor how icons are loaded 😋